### PR TITLE
marqeta-client-user - user module with list() function

### PR DESCRIPTION
**Pull request by jbearden**

### CODE REVIEW

File: src/user.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  birthDate?:string,
+  address1?: string;
+  city?: string;
+  state?: string;
+  country?: string;
+  postalCode?: string;
+  phone?:string;
+  gender?:string,
+  usesParentAccount?: boolean,
+  metadata: any
```
Comment (drbobbeaty) [inline src/user.ts:32]: You ended almost all but the last three lines with semi-colons `;`... in my `interface` definitions, I've used semi-colons on all lines - even the last, which is something to consider... but here, I'd just say pick a way to handle this, and stick with it for the entire project.

File: src/user.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  } = {}): Promise<{
+    success: boolean,
+    body?: UserList,
+    error?: MarqetaError,
+  }> {
+    const searchOptions = snakecaseKeys(search)
+    const resp = await this.client.fire('GET',
+      'users',
+      { ...searchOptions }
+    )
```
Comment (drbobbeaty) [inline src/user.ts:71]: Why don't you just say:

```typescript
     const resp = await this.client.fire('GET',
       'users',
       snakecaseKeys(search),
     )
```
and be done with it?

**Review by drbobbeaty — CHANGES_REQUESTED** [review]
A couple of minor things...

File: src/user.ts (hunk 1 above)
Comment (jbearden) [reply src/user.ts:32]: oh wow, and `npx tsc` nor `eslint` picked that up!

File: src/user.ts (hunk 2 above)
Comment (jbearden) [reply src/user.ts:71]: yes, good catch.  Added test for count paramter too.

**Review by drbobbeaty — APPROVED** [review]
When you are happy with it... I am.

**Pull request merged**